src/app/containers/MediaPlayer/index.jsx

```diff
@@ -168,40 +174,45 @@ const MediaPlayerContainer = ({
     translations,
   );
 
-  const renderCaption = () =>
-    captionBlock ? (
-      <Caption block={captionBlock} type={mediaInfo.type} service={service} />
-    ) : null;
+  const renderCaption = captionBlock ? (
+    <Caption block={captionBlock} type={mediaInfo.type} service={service} />
+  ) : null;
+
+  const mediaPlayer = isAmp ? (
+    <AmpMediaPlayer
+      src={embedSource}
+      placeholderSrc={placeholderSrc}
+      placeholderSrcset={placeholderSrcset}
+      title={iframeTitle}
+      noJsMessage={translatedNoJSMessage}
+      service={service}
+    />
+  ) : (
+    <CanonicalMediaPlayer
+      src={embedSource}
+      placeholderSrc={placeholderSrc}
+      placeholderSrcset={placeholderSrcset}
+      showPlaceholder={showPlaceholder}
+      title={iframeTitle}
+      service={service}
+      mediaInfo={mediaInfo}
+      noJsMessage={translatedNoJSMessage}
+      noJsClassName="no-js"
+      showLoadingImage={showLoadingImage}
+    />
+  );
 
   return (
     <>
       <Metadata aresMediaBlock={aresMediaBlock} embedSource={embedSource} />
-      <Figure>
-        {isAmp ? (
-          <AmpMediaPlayer
-            src={embedSource}
-            placeholderSrc={placeholderSrc}
-            placeholderSrcset={placeholderSrcset}
-            title={iframeTitle}
-            noJsMessage={translatedNoJSMessage}
-            service={service}
-          />
-        ) : (
-          <CanonicalMediaPlayer
-            src={embedSource}
-            placeholderSrc={placeholderSrc}
-            placeholderSrcset={placeholderSrcset}
-            showPlaceholder={showPlaceholder}
-            title={iframeTitle}
-            service={service}
-            mediaInfo={mediaInfo}
-            noJsMessage={translatedNoJSMessage}
-            noJsClassName="no-js"
-            showLoadingImage={showLoadingImage}
-          />
-        )}
-        {showCaption && renderCaption()}
-      </Figure>
+      {showCaption && renderCaption ? (
+        <Figure>
+          {mediaPlayer}
+          {showCaption && renderCaption}
```

Comment:
```suggestion
          {showCaption && caption}
```

Reply:
Thanks @jroebu14 for the detailed review 👍. Also these small renaiming have been taken care of.